examples/02_decoding/plot_haxby_grid_search.py

```diff
@@ -15,120 +14,135 @@
-# Here we use a Support Vector Classification, with a linear kernel
-from sklearn.svm import SVC
-svc = SVC(kernel='linear')
-
-
-# Define the dimension reduction to be used.
-# Here we use a classical univariate feature selection based on F-test,
-# namely Anova. We set the number of features to be selected to 500
-from sklearn.feature_selection import SelectKBest, f_classif
-feature_selection = SelectKBest(f_classif, k=500)
-
-# We have our classifier (SVC), our feature selection (SelectKBest), and now,
-# we can plug them together in a *pipeline* that performs the two operations
-# successively:
-from sklearn.pipeline import Pipeline
-anova_svc = Pipeline([('anova', feature_selection), ('svc', svc)])
+# ANOVA pipeline with :class:`nilearn.decoding.Decoder` object
+# ------------------------------------------------------------
+#
+# Nilearn Decoder object aims to provide smooth user experience by acting
+# as a pipeline of several tasks: preprocessing with NiftiMasker, decoding
+# with different types of estimators (in this example is Support Vector
+# Machine with a linear kernel) on nested cross-validation, selecting
+# relevant features with ANOVA -- a classical univariate feature selection
+# based on F-test.
+from nilearn.decoding import Decoder
+# Here screening_percentile is set to 2 percent, meaning around 800
+# features will be selected with ANOVA.
+decoder = Decoder(estimator='svc', cv=5, mask=mask_filename,
+                  smoothing_fwhm=4, standardize=True,
+                  screening_percentile=2)
 
 ###########################################################################
-# Compute prediction scores using cross-validation
+# Fit the Decoder and predict the reponses
 # -------------------------------------------------
-anova_svc.fit(X, y)
-y_pred = anova_svc.predict(X)
-
-from sklearn.model_selection import cross_val_score
+# As a complete pipeline by itself, decoder will perform cross-validation
+# for the estimator, in this case Support Vector Machine. We can output the
+# best parameters selected for each cross-validation fold. See
+# https://scikit-learn.org/stable/modules/cross_validation.html for an
+# excellent explanation of how cross-validation works.
+#
+# First we fit the Decoder
+decoder.fit(fmri_niimgs, y)
+for i, (param, cv_score) in enumerate(zip(decoder.cv_params_['shoe']['C'],
+                                          decoder.cv_scores_['shoe'])):
+    
+    print("Fold %d | Best SVM parameter: %.1f with score: %.3f" % (i + 1,
+          param, cv_score))
+# Output the prediction with Decoder
+y_pred = decoder.predict(fmri_niimgs)
 
-k_range = [10, 15, 30, 50, 150, 300, 500, 1000, 1500, 3000, 5000]
+###########################################################################
+# Compute prediction scores with different values of screening percentile
+# -----------------------------------------------------------------------
+import numpy as np
+screening_percentile_range = [2, 4, 8, 16, 32, 64]
```

Comment:
The `screening_percentile` is not that flexible compare to specifying a fix number of voxels because `screening_percentile` must be an integer between `[1, 100]`.

For example if the number of voxels is 100,000 we cannot choose less than 1000 voxels (1 percent) with `screening_percentile`

Reply:
Ok. Thanks for the details. I missed this thinking that screening percentile is passed to ANOVA.

Reply:
I think more flexibility with screening percentile is not a priority right now, we should leave it to future enhancements if needed by someone

Reply:
yes i understand